app-connector-client/server/app.js

```diff
@@ -12,27 +12,24 @@ var expressWinston = require('express-winston');
 
 //route definitions
 const events = require("./routes/events")
-var connector = require("./routes/connector")
+var connector;
 var apis = require("./routes/apis")
 var keys = require("./keys")
 
 var app = express()
 var varkesConfig
 
-
-
-
-
 module.exports = function (varkesConfigPath = null, nodePortParam = null) {
     nodePort = nodePortParam;
 
     app.use(bodyParser.json());
 
     if (varkesConfigPath) {
-        endpointConfig = path.resolve(varkesConfigPath)
-        LOGGER.info("Using configuration %s", endpointConfig)
-        varkesConfig = require(endpointConfig)
+        endpointConfig = path.resolve(varkesConfigPath);
+        LOGGER.info("Using configuration %s", endpointConfig);
+        varkesConfig = require(endpointConfig);
         configValidation(varkesConfig)
+        connector = require("./routes/connector")(varkesConfig, nodePortParam);
```

Comment:
but I need to pass the config file and the node port with the initialization of the module. Although it's passed by reference and passing the variable before being initialized could thoertically work, I think it will be more messy

Reply:
Can't you do in the variables section something like:
connectorModule = require("./routes/connector")

and then later in the program you can call:
connector = connectorModule(varkesConfig, nodePortParam))

Reply:
done